sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -46,6 +46,7 @@ case class Exchange(newPartitioning: Partitioning, child: SparkPlan) extends Una
     child.sqlContext.sparkContext.conf.getInt("spark.shuffle.sort.bypassMergeThreshold", 200)
 
   override def execute(): RDD[Row] = attachTree(this , "execute") {
+    lazy val sparkConf = Option(SparkEnv.get).map(_.conf).getOrElse(new SparkConf())
```

Comment:
All nodes should have a `sqlContext` which should have a `sparkContext`.  Can we just get this from there instead?